### PR TITLE
Added separated channel for each producer and consumer.

### PR DESCRIPTION
**Pull request by cheshir**
Also fixed case when non-durable queues and exchanges was not recreated after reconnect.

### CODE REVIEW

File: consumer.go [hunk 1]
```diff
@@ -64,40 +66,73 @@ func (consumer *consumer) Stop() {
...
-func (worker worker) Run(handler ConsumerHandler) {
+func (worker *worker) Run(handler ConsumerHandler) {
 	atomic.StoreInt32(&worker.status, statusRunning)
 
 	for {
 		select {
 		case message := <-worker.deliveries:
-			if message == nil { // Channel has been closed.
-				return
+			if message == nil { // Seems like channel was closed.
```
Comment (ynotLeft) [inline consumer.go]: **It** seems like channel was closed.

File: consumer.go [hunk 2]
```diff
@@ -64,40 +66,73 @@ func (consumer *consumer) Stop() {
...
+				if worker.changeStatusToStoppedAtomic() {
```
Comment (ynotLeft) [inline consumer.go]: boolean should answer yes/no question so 
worker.**hasChangedStatusToStoppedAtomic**()

File: consumer.go [hunk 3]
```diff
@@ -64,40 +66,73 @@ func (consumer *consumer) Stop() {
...
+					// Stop the worker.
+
+					return
+				} else {
+					// Somebody already trying to stop the worker.
```
Comment (ynotLeft) [inline consumer.go]: Somebody **is** already trying to stop the worker.

File: mq.go [hunk 4]
```diff
@@ -21,26 +20,40 @@ const (
...
+
+type mq struct {
+	channel              wabbit.Channel
+	config               Config
+	connection           wabbit.Conn
+	errorChannel         chan error
+	internalErrorChannel chan error
+	consumers            *consumersRegistry
+	producers            *producersRegistry
+	reconnectStatus      int32 // Defines whether client is trying to reconnect or not.
```
Comment (ynotLeft) [inline mq.go]: Consider simplifying to: 
Defines if client is trying to reconnect.

File: mq.go [hunk 5]
```diff
@@ -49,52 +62,158 @@ func New(config Config) (MQ, error) {
...
-	GetConsumer(name string) (Consumer, error)
-	SetConsumerHandler(name string, handler ConsumerHandler) error
-	GetProducer(name string) (Producer, error)
-	Error() <-chan error
-	Close()
+// GetConsumer returns a consumer by its name or error if consumer wasn't found.
+func (mq *mq) GetConsumer(name string) (consumer Consumer, err error) {
+	consumer, ok := mq.consumers.Get(name)
+	if !ok {
+		err = fmt.Errorf("Conumer '%s' is not registered. Check your configuration.", name)
```
Comment (ynotLeft) [inline mq.go]: Conumer -> Consumer

File: mq.go [hunk 6]
```diff
@@ -49,52 +62,158 @@ func New(config Config) (MQ, error) {
...
+		mq.internalErrorChannel <- err
+	}
+}
+
+func (mq *mq) errorHandler() {
+	for err := range mq.internalErrorChannel {
+		select {
+		case mq.errorChannel <- err: // Proxies errors to the user.
+		default: // Drop errors if channel buffer is full.
+			// TODO It probably makes sense to make it optional or even remove.
```
Comment (ynotLeft) [inline mq.go]: TODO It probably makes sense to make it optional or even remove **it**.

**Comment by cheshir** [comment]
@ynotLeft thanks a lot!

**Pull request merged**